src/libraries/System.IO.FileSystem.AccessControl/tests/FileSystemAclExtensionsTests.cs

```diff
@@ -177,79 +177,65 @@ public void DirectoryInfo_Create_NullDirectoryInfo()
-            DirectoryInfo info = new DirectoryInfo(path);
-            DirectorySecurity security = new DirectorySecurity();
+            var directory = new TempAclDirectory();
+            string path = Path.Combine(directory.Path, "ParentDoesNotExist");
+            directory.Dispose(); // Delete parent folder
 
-            Assert.Throws<UnauthorizedAccessException>(() =>
-            {
-                if (PlatformDetection.IsNetFramework)
-                {
-                    FileSystemAclExtensions.Create(info, security);
-                }
-                else
-                {
-                    info.Create(security);
-                }
-            });
+            DirectoryInfo info = new DirectoryInfo(path);
+            DirectorySecurity security = GetDirectorySecurity(FileSystemRights.FullControl);
+            DirectoryInfo_Create_Framework(info, security);
         }
 
-        [Fact]
-        public void DirectoryInfo_Create_DefaultDirectorySecurity()
+        private void DirectoryInfo_Create_Framework(DirectoryInfo info, DirectorySecurity security)
         {
-            DirectorySecurity security = new DirectorySecurity();
-            Verify_DirectoryInfo_Create(security);
+            if (PlatformDetection.IsNetFramework)
+            {
+                FileSystemAclExtensions.Create(info, security);
+            }
+            else
+            {
+                info.Create(security);
+            }
         }
 
         [Theory]
-        [InlineData(FileSystemRights.ReadAndExecute, AccessControlType.Allow)]
-        [InlineData(FileSystemRights.ReadAndExecute, AccessControlType.Deny)]
-        [InlineData(FileSystemRights.WriteData,      AccessControlType.Allow)]
-        [InlineData(FileSystemRights.WriteData,      AccessControlType.Deny)]
-        [InlineData(FileSystemRights.FullControl,    AccessControlType.Allow)]
-        [InlineData(FileSystemRights.FullControl,    AccessControlType.Deny)]
-        public void DirectoryInfo_Create_DirectorySecurityWithSpecificAccessRule(
-            FileSystemRights rights,
-            AccessControlType controlType)
-        {
-            DirectorySecurity security = GetDirectorySecurity(rights, controlType);
-            Verify_DirectoryInfo_Create(security);
+        // Must have at least one Read, otherwise the TempAclDirectory will fail to delete that item on dispose
+        [InlineData(FileSystemRights.FullControl)]
+        [InlineData(FileSystemRights.Read)]
+        [InlineData(FileSystemRights.Read | FileSystemRights.Write)]
+        [InlineData(FileSystemRights.Read | FileSystemRights.Write | FileSystemRights.ExecuteFile)]
+        [InlineData(FileSystemRights.ReadAndExecute)]
+        [InlineData(FileSystemRights.ReadAttributes | FileSystemRights.ReadData | FileSystemRights.ReadPermissions)]
+        public void DirectoryInfo_Create_DirectorySecurityWithSpecificAccessRule(FileSystemRights rights)
```

Comment:
`WriteData` is covered by the `Write` flag:

https://github.com/dotnet/runtime/blob/b23ccde65bff6b0964371cc47967af162ec5f2df/src/libraries/System.IO.FileSystem.AccessControl/src/System/Security/AccessControl/FileSystemRights.cs#L42

Which reminds me that the last `InlineData` is unnecessary, since they are all already covered by `FileSystemRights.Read`.

Reply:
For #2, I added two new unit tests that combine testing both `Allow` and `Deny`. One for `DirectoryInfo` and another one for `FileInfo`. Can you please take a look?

Reply:
I brought back all the FileSystemRights that were testable.